frontend/lib/wallets/MetaMask.ts

```diff
@@ -41,7 +55,20 @@ export default class MetaMask extends AbstractWallet {
     }
 
     getProvider(): ethers.providers.JsonRpcProvider {
-        return new ethers.providers.Web3Provider(window.ethereum, 'any');
+        if (!MetaMask.provider) {
+            MetaMask.provider = window.ethereum;
+            // handle case of MetaMask and Coinbase Wallet extensions both installed
+            if (typeof window.ethereum !== 'undefined') {
+                if (window.ethereum.providers?.length) {
+                    window.ethereum.providers.forEach((p: any) => {
+                        if (p.isMetaMask) {
+                            MetaMask.provider = p;
+                        }
+                    });
```

Comment:
if there's supposed to be just 1 metamask object in this array, you can stop the loop after the `if` condition is hit.